src/query/sql/src/planner/optimizer/property/selectivity.rs

```diff
@@ -221,122 +222,14 @@ impl<'a> SelectivityEstimator<'a> {
-                        } else {
-                            return Ok(DEFAULT_SELECTIVITY);
-                        };
-                        // Greater than or equal to predicate is similar to greater than predicate.
-                        let mut num_greater = 0.0;
-                        let new_min = const_datum.clone();
-                        let new_max = column_stat.max.clone();
-                        for bucket in col_hist.buckets_iter() {
-                            if let Ok(ord) = bucket.upper_bound().compare(&const_datum) {
-                                if ord == Ordering::Less {
-                                    num_greater += bucket.num_values();
-                                } else {
-                                    break;
-                                }
-                            } else {
-                                return Ok(DEFAULT_SELECTIVITY);
-                            }
-                        }
-                        let selectivity = 1.0 - num_greater / col_hist.num_values();
-                        if update {
-                            update_statistic(column_stat, new_min, new_max, selectivity)?;
-                            self.updated_column_indexes.insert(column_ref.column.index);
-                        }
-                        Ok(selectivity)
-                    }
-                    ComparisonOp::LTE => {
-                        let col_hist = if let Some(hist) = column_stat.histogram.as_ref() {
-                            hist
-                        } else {
-                            return Ok(DEFAULT_SELECTIVITY);
-                        };
-                        // Less than or equal to predicate is similar to less than predicate.
-                        let mut num_greater = 0.0;
-                        let new_max = const_datum.clone();
-                        let new_min = column_stat.min.clone();
-                        for bucket in col_hist.buckets_iter() {
-                            if let Ok(ord) = bucket.upper_bound().compare(&const_datum) {
-                                if ord == Ordering::Less || ord == Ordering::Equal {
-                                    num_greater += bucket.num_values();
-                                } else {
-                                    break;
-                                }
-                            } else {
-                                return Ok(DEFAULT_SELECTIVITY);
-                            }
-                        }
-                        let selectivity = num_greater / col_hist.num_values();
-                        if update {
-                            update_statistic(column_stat, new_min, new_max, selectivity)?;
-                            self.updated_column_indexes.insert(column_ref.column.index);
-                        }
-                        Ok(selectivity)
-                    }
+                    _ => Self::compute_binary_comparison_selectivity(
```

Comment:
It seems that if `left` is `ScalarExpr::ConstantExpr(_)`, we should use `op.reverse()` instead of `op`(this was a previous bug).

Reply:
Miss a test for this case?

Reply:
The previous test can cover this case, but InferFilterOptimizer puts the ConstantExpr in the simple predicate to the right side, which makes it pass the previous test. Maybe we can add a test `explain select * from tt where 10 < number or 20 < number;` to cover this case.
main:
```sql
create table t as select * from numbers(100);
explain select * from t where 10 < number or 20 < number;

-[ EXPLAIN ]-----------------------------------
Filter
├── output columns: [t.number (#0)]
├── filters: [10 < t.number (#0) OR 20 < t.number (#0)]
├── estimated rows: 27.74
└── TableScan
    ├── table: default.default.t
    ├── output columns: [number (#0)]
    ├── read rows: 100
    ├── read size: < 1 KiB
    ├── partitions total: 1
    ├── partitions scanned: 1
    ├── pruning stats: [segments: <range pruning: 1 to 1>, blocks: <range pruning: 1 to 1>]
    ├── push downs: [filters: [10 < t.number (#0) OR 20 < t.number (#0)], limit: NONE]
    └── estimated rows: 100.00
```

Reply:
Yes